src/norm/model.nim

```diff
@@ -114,3 +114,25 @@ proc checkRo*(T: typedesc[Model]) =
   when T.hasCustomPragma(ro):
     {.error: "can't use mutating procs with read-only models".}
 
+proc getRelatedFieldNameOn*[T: Model, M: Model](targetModel: typedesc[T], sourceModel: typedesc[M]): string  {.raises: [FieldDefect].} =
+  for sourceFieldName, sourceFieldValue in sourceModel()[].fieldPairs:
```

Comment:
This contains another issue in that it won't deliver you what you'd expect if M has 2 foreign key fields to T. This doesn't let you properly pick the field if there are 2 fields and with the logic I provided that isn't exactly possible.
I stumbled over this issue a couple weeks ago during implementation in my nimstoryfont project where I already work with these procs.

The way around that is overloading "getManyFromOne" and "getManyToMany" so that you can pass them the name of the field you want to fetch stuff for. This will be included in a fairly large update to this PR where I'll also take your comments into account.

Reply:
1. Done
2. I went with "target" and "source", for I generally try to avoid using shorthands
3. Done.
4. I renamed it to T,S for target and source accordingly.